pkg/vadmin/interface.go

```diff
@@ -103,14 +108,64 @@ func MakeAdmintools(log logr.Logger, vdb *vapi.VerticaDB, prunner cmds.PodRunner
 // vclusterops library to perform all of the admin operations via RESTful
 // interfaces.
 type VClusterOps struct {
-	Log logr.Logger
-	VDB *vapi.VerticaDB
+	Log    logr.Logger
+	VDB    *vapi.VerticaDB
+	Client client.Client
+	VClusterOpsInterface
 }
 
 // MakeVClusterOps will create a dispatcher that uses the vclusterops library for admin commands.
-func MakeVClusterOps(log logr.Logger, vdb *vapi.VerticaDB) Dispatcher {
+func MakeVClusterOps(log logr.Logger, vdb *vapi.VerticaDB, cli client.Client, vopsi VClusterOpsInterface) Dispatcher {
 	return VClusterOps{
-		Log: log,
-		VDB: vdb,
+		Log:                  log,
+		VDB:                  vdb,
+		Client:               cli,
+		VClusterOpsInterface: vopsi,
 	}
 }
+
+type HTTPSCerts struct {
+	Key    string
+	Cert   string
+	CaCert string
+}
+
+// retrieve the certs from HTTPServerTLSSecret for calling NMA endpoints
+func (v VClusterOps) retrieveHTTPSCerts(ctx context.Context) (*HTTPSCerts, error) {
```

Comment:
Can we move this function to a separate file? Kind of like at.go, but for VClusterOps. Maybe call it vc.go.

Reply:
Will do it.